### DISCUSSION TITLE
Are These Features of Interest for Upstreaming?

### DISCUSSION THREAD

**Discussion by kiancross** [pretix/pretix#4725] [Ideas]
I’ve made some changes to a fork of Pretix for another project. While I understand that not all features may be suitable for upstreaming, I believe a couple might be worth considering:

### Order Expiry Warning Granularity

* Adjusted the order expiry warnings to work on an hourly basis instead of daily.
* This is particularly useful for cases where the order expiry is set to a short duration, allowing you to send a reminder email, for instance, 12 hours before the expiry.

### Order Page Access Restriction:

* When a logged-in user places an order, the link to the order page can only be accessed when logged into the associated account.
* This prevents unauthorised access, such as when links in forwarded emails are shared, ensuring that changes to the order can only be made by someone with account access.

Please let me know if either of these features would be of interest for upstreaming.

**Comment by raphaelm** [comment]
Thanks for offering, this is a nice way of asking :)

> Order Expiry Warning Granularity

I'm leaning towards "no", with the following reasoning. In our experience, 99% of pretix shops fall into one of two categories:

1. Payment deadline in the area of 5-20 days. Here, a warning with day granularity is absolutely fine.
2. Payment deadline in the area of 15-60 minutes (basically: instant payment). Here, I think a payment reminder is entirely out of place – people will not have forgotten their payment minutes after the sale. The only reason they need to be reminded is when the payment unexpectedly fails (possibly after they closed the page). This is a job for the "payment failed" email, though.

Therefore, it feels like unnecessary complexity.

Also, the reminders are handled by cronjob and most pretix setups do not make guarantees to run cronjobs more often than every 10-30 minutes, so I'd see a risk of people relying on these reminders even if it's not guarenteed that they are sent on time.

> Order Page Access Restriction

Yes, I think we'd be interested in that feature if it's opt-in (or opt-out?). There would probably need to be some discussion on where and how this should be best configured – probably in organizer-level settings as part of the customer account settings.

**Reply by kiancross** [reply]
> In our experience, 99% of pretix shops fall into one of two categories.

No problem! It sounds like we're in the minority with a 24-hour payment window, for which hour-granularity is useful.

> Yes, I think we'd be interested in that feature if it's opt-in (or opt-out?). There would probably need to be some discussion on where and how this should be best configured – probably in organizer-level settings as part of the customer account settings.

How about putting a checkbox in the organiser-level 'Customer Accounts' settings with the following text:

> > Require login to access order URLs. This setting only applies to orders purchased while the user was logged in.

In the future, the feature could be made more granular so that customers can choose themselves whether or not they want to protect their order URLs in this way.

**Reply by raphaelm** [reply]
> How about putting a checkbox in the organiser-level 'Customer Accounts' settings with the following text:
>>  Require login to access order URLs. This setting only applies to orders purchased while the user was logged in.

Yup, sounds good!

> In the future, the feature could be made more granular so that customers can choose themselves whether or not they want to protect their order URLs in this way.

I'm not sure that's required, but sure, we can always discuss it later :)
